### PR TITLE
[BLD] [CLN] Close assorted issues - bare exceptions, unused func

### LINKED ISSUE #19676: Unused util
util._doctools.TablePlotter is used only in it's own `__main__` block.  Does that need to exist?

_decorators.docstring_wrapper is unused.

### LINKED ISSUE #21991: BLD: Make CLI option to enable coverage
https://github.com/pandas-dev/pandas/pull/21879#discussion_r204037952

### PR DESCRIPTION
**Pull request by jbrockmendel**
<s>- Removing numpy cimport from cython modules where feasible fixes the npy_deprecated1.7 ... warning.</s> <b>update</b> Nope!
- Catch specific exceptions in a handful of places.
- Removes unused function #19676
- Implement command-line option to enable cython coverage #21991

closes #19676
closes #21991
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Review by gfyoung — APPROVED** [review]
Part of me wants to make this more modular, but the changes are small enough that this isn't an issue.  LGTM!

File: pandas/util/_decorators.py [hunk 1]
```diff
@@ -339,48 +338,3 @@ def make_signature(func):
     if spec.keywords:
         args.append('**' + spec.keywords)
     return args, spec.args
-
-
-class docstring_wrapper(object):
-    """
```
Comment (jreback) [inline pandas/util/_decorators.py]: yeah was going to use this a while back, but didn't need

**Comment by jreback** [comment]
hmm still quite a few warnings (I am guessing that this AND the other PR will make progress on more of these?)

ideally we could run the --with-cython-coverage on a build at some point. though would have to find a way to display this.

**Comment by jreback** [comment]
lgtm. can you rebase again, just want to see what kinds of warnings on build travis has now (this is the last in the series to remove warnings?)

**Comment by jbrockmendel** [comment]
Will rebase in a bit, ok.  But note that the build-related parts of the PR were reverted they broke on windows.

**Comment by jreback** [comment]
thank @jbrockmendel

**Comment by jreback** [comment]
@jbrockmendel build warnings are almost gone now! thanks!

**Pull request merged**